lms/static/js/groups/views/cohort_discussions.js

```diff
@@ -0,0 +1,208 @@
+                        isCategoryCohorted: isCategoryCohorted
+                    });
+                }
+                return html;
+            }, this).join('');
+        },
+        toggleDisabledAttribute: function($element, enable) {
+            if (enable) {
+                $element.prop('disabled', false);
+            } else {
+                $element.prop('disabled', 'disabled');
+            }
+        },
+        changeDiscussionCourseWideCategory: function(event) {
+            event.preventDefault();
+            this.toggleDisabledAttribute(this.$('.cohort-course-wide-discussions-form .action-save'), true);
+        },
+        changeAllInlineDiscussions: function(event) {
+            event.preventDefault();
+            this.toggleInlineDiscussions(!($(event.currentTarget).prop('checked')));
+        },
+        changeCohortInlineDiscussions: function(event) {
+            event.preventDefault();
+            this.toggleInlineDiscussions(($(event.currentTarget).prop('checked')));
+        },
+        toggleInlineDiscussions: function(enable) {
+            this.toggleDisabledAttribute(this.$('.check-discussion-category'), enable);
+            this.toggleDisabledAttribute(this.$('.check-discussion-subcategory-inline'), enable);
+            this.toggleDisabledAttribute(this.$('.cohort-inline-discussions-form .action-save'), true);
+        },
+        changeDiscussionInlineCategory: function(event) {
+            var $selectedCategory = $(event.currentTarget);
+
+            if (!$selectedCategory.prop('checked')) {
+                $('.check-all-inline-discussions').prop('checked', false);
+            }
+        },
+        changeDiscussionInlineSubCategory: function (event) {
+            var $selectedTopic = $(event.currentTarget);
+            if (!$selectedTopic.prop('checked')) {
+                $('.check-all-inline-discussions').prop('checked', false);
+            }
+            this.toggleDisabledAttribute(this.$('.cohort-inline-discussions-form .action-save'), true);
+        },
+        getCohortedDiscussions: function(selector) {
+            var self=this;
+            self.cohortedDiscussionTopics = [];
+            _.each(self.$(selector), function (topic) {
+                self.cohortedDiscussionTopics.push($(topic).data('id'))
+            });
+        },
+        saveInlineDiscussionsForm: function (event) {
+            event.preventDefault();
+
+            var self = this;
+            self.getCohortedDiscussions('.check-discussion-subcategory-inline:checked');
+
+            this.cohortSettings.set({
+                cohorted_inline_discussions: self.cohortedDiscussionTopics,
+                always_cohort_inline_discussions: self.$('.check-all-inline-discussions').prop('checked')
+            });
+
+            self.saveForm(self.$('.inline-discussion-topics'))
+                .done(function () {
+                    self.model.fetch().done(function () {
+                        self.render();
```

Comment:
If I understand this correctly, doing this will re-render both the forms, right? And if so, the changes in the other form will be lost? Can we break the two forms into separate views so that only one is re-rendered?

"Working With Nested Models Or Collections" on http://addyosmani.github.io/backbone-fundamentals/ might be useful.